extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +336,87 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    const TIME_WAIT = 500;
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
+    this.debouncedBoundOnSession_ = debounce(
+      this.root.ampdoc.win,
+      this.boundOnSession_,
+      TIME_WAIT
+    );
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(BrowserEventType).forEach((key) => {
+      root.removeEventListener(BrowserEventType[key], this.boundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.observables_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    userAssert(
+      isExperimentOn(this.root.ampdoc.win, 'custom-browser-event-tracker'),
+      'expected global "custom-browser-event-tracker" experiment to be enabled'
+    );
+
+    const selector = userAssert(
+      config['selector'],
+      'Missing required selector on browser event trigger'
+    );
+    userAssert(
+      selector.length,
+      'Missing required selector on browser event trigger'
+    );
+    assertUniqueSelectors(selector);
+
+    const selectionMethod = config['selectionMethod'] || null;
+    const eventName = config['on'];
```

Comment:
1. Let's not spill the concept of discerning by "browser events" into the user layer. Instead, let's specify the event itself on the user warnings.
2. Let's also use a single assertion for presence and length.
3. nit for style in `|| null` and destructure.

```suggestion
    const {
      'selectionMethod': selectionMethod = null,
      'on': eventName,
      'selector': selector,
    } = config;
    userAssert(
      selector?.length,
      `Missing required selector on configuration for trigger on "${on}"`
    );
    assertUniqueSelectors(selector);
```